### ISSUE TITLE
Multiple PRs and behavior of API readiness app and SDK release app 

### ISSUE THREAD

**Issue by ladonnaq** [Azure/azure-sdk-tools#6046]
I have noticed in working with service teams using their real data that it is not always clear as to which PR they should use for API readiness.  Also, the PR they use for API Readiness is not the same PR they use to request release of the SDKs.  The PR provided to release the SDKs does not always meet API readiness (i.e. missing approvals). 

@josefree Would you please review and provide feedback?  This scenario does not follow the simple path of PR in private repo that meets all validations/approvals and then that PR is published to public repo and is used to generate the SDKs.  

Here is an example using  AFOI-Network Cloud data.  This is a new service and this is the initial release of their beta SDKs.  

- This is the PR they provided for me to test API readiness for public preview/initial beta SDK release - https://github.com/Azure/azure-rest-api-specs/pull/23086
- This is also the PR that they used to request the SDK releases and the one that they provided to me to test the release planner.  
- However, this PR does not pass API readiness.  The PR does have the label Approved-BreakingChanges but the API readiness app is not recognizing this and is asking the user to request suppression.  This blocks ARM review status.  However, if the breaking changes approval were to pass, there is no label on this PR for ARMSignedOff so not sure it would pass ARM review status.  
![image](https://user-images.githubusercontent.com/67656762/234389982-2729b06e-496c-4d68-9fb8-942b7e9496c3.png)
![image](https://user-images.githubusercontent.com/67656762/234391270-bd7a0212-ddd3-4386-bffd-ac16bad8ae7b.png)

- This is the PR that created when the API Spec was initially published from the private repo to the public repo - https://github.com/Azure/azure-rest-api-specs/pull/22560.  This is typically the PR we would expect that would be used to release the SDKs (the one that should meet API readiness). It passes all of the checks except ARM Review because there is no ARMSignedOff label on this PR.  
- Instead, there is a list of PRs in the private repo that have various approvals.  
The RPs approved and merged into azure-rest-api-specs-pr:
https://github.com/Azure/azure-rest-api-specs-pr/pull/10272
https://github.com/Azure/azure-rest-api-specs-pr/pull/10535
https://github.com/Azure/azure-rest-api-specs-pr/pull/10890
https://github.com/Azure/azure-rest-api-specs-pr/pull/10985
https://github.com/Azure/azure-rest-api-specs-pr/pull/11199

-  This only PR in the private repo that passes API readiness -https://github.com/Azure/azure-rest-api-specs-pr/pull/10985. 
![image](https://user-images.githubusercontent.com/67656762/234396632-16dc5322-edb8-4d0a-9abf-3ccbda18c53f.png)

**Comment by josefree** [comment]
In release planner, they should go back to update the release plan - update the target release date or associated the spec pull request.  So that to ensure the SDK release will catch up their latest updates to the API version.

**Comment by ladonnaq** [comment]
> In release planner, they should go back to update the release plan - update the target release date or associated the spec pull request. So that to ensure the SDK release will catch up their latest updates to the API version.

Your comments make sense, but using the Release Planner apps the user would not be successful.  

In the example of AFOI-Network Cloud: 
**API Readiness for Public Preview (API Spec is still in the private specs repo).** 
1.  The PR in the private repo for the first published API version is - https://github.com/Azure/azure-rest-api-specs-pr/pull/12105/files.  This is the PR I would expect that the team would be using to validate API readiness for Public Preview (because the PR has not been published to the public repo yet).  
2. The PR fails API readiness, even though if I go look at the PR I can manually validate it has been approved and already merged.  Screenshot below showing the failure.  The service team requested suppression and it was approved (you can see this in the screenshot and in the details of the PR).  However, later on the flow of the API Readiness app this is blocking ARM Review status of approved and sign-off.  
![image](https://user-images.githubusercontent.com/67656762/234669121-98370997-86e7-4ed6-95c0-e8f751ce9d08.png)
4. For some reason the generation of the API Ref document is not working in the API Readiness App and when I look in the PR the message states that it has been in this state for over 6 days (trying to generate the API Ref Doc preview).  
![image](https://user-images.githubusercontent.com/67656762/234670298-318baff5-4a36-47db-8c7e-6f6e4752ad8b.png)
5. ARM review fails because of the validation error earlier.  
![image](https://user-images.githubusercontent.com/67656762/234670525-7cb9cc1f-2065-4a15-af9c-9eec29be4a7a.png)
6. The signoff page shows the validation error and blocks from completing API Readiness.  
![image](https://user-images.githubusercontent.com/67656762/234672726-14aea18c-2ef6-405b-8f86-1cd62accc234.png)

**Let's assume the PR above did pass API Readiness.**  The service team would move on to the SDK Release App.  In the current implementation of the Release Planner if a service team wants to change the PR (which already met API Readiness) for SDK release, they are instructed to return to the API Readiness Application.  However, if the service team uses the latest PR it may not pass API Readiness.  This is the case with AFOI-Network Cloud.  The latest PR is the one that they provided when they requested release of the SDKs via OpenAPI Hub.  This does not pass API Readiness (as referenced in the description of this GitHub issue). 

What are your thoughts on resolving this issue? I expect this scenario to be common and right now the release planner implementation is incorrect (users would be blocked from meeting API Readiness and would therefore be blocked from releasing their SDKs using the SDK release app).  
- I think that there are unexpected bugs in the API readiness app back-end.  I will open an issue for engineering to investigate.  The PR in the private repo has met API readiness if you look at the PR.  Also, the preview doc issue.  That may be a bug, not sure. 
- In the case of the PR that the service team associates with the SDK release app for release.  If it is public preview, then the PR used for API Readiness could have been in the private repo.  They may not publish it until SDK release.  So the user would have to associate another PR (one in the public repo) for SDK release. This PR may or may not pass API Readiness.  In the case of AFOI-Network Cloud it does not.  Would it make more sense to NOT have a service team go back to API Readiness to validate the API again if it has already passed API Readiness?  If so, is there any risk in doing so?  If we don't have the user return to API readiness, then we would allow them to associate a new PR in the SDK release app that is for SDK release not API readiness.  Is there any validations that would be required for PR for SDK release?  Is it enough to check to see if it merged?  (I would expect if there are errors merging would be blocked).

**Comment by josefree** [comment]
![image](https://user-images.githubusercontent.com/18413981/234778589-a1b20df7-cf47-47b2-9096-2f8bbfc61398.png)
If the Avacado validation failure is approved, the it should not block ARM review and approval later.  This might be an issue to fix for the release planner

**Comment by josefree** [comment]
![image](https://user-images.githubusercontent.com/18413981/234779084-ea8b4e1c-d91d-49f9-929f-2b46e669a6dc.png)
This has to be escalated to Tooling team to take a look. But very likely they will need SKILL team help.  Terry Jin of SKILL team used to be the go-to person for this hang-on issue.

**Comment by josefree** [comment]
![image](https://user-images.githubusercontent.com/18413981/234779546-6345011f-a409-4cab-be72-f805e4ae9517.png)
we should create an issue to fix the problem. When there is validation error but approved, the step should be unblocked.

**Comment by josefree** [comment]
![image](https://user-images.githubusercontent.com/18413981/234779783-61dc2e01-259c-4bf7-9662-ac24be21697e.png)
We need an issue to fix the problem, when there is a spec validation failure but approved to neglect, the status should be "Failed with approval". and the signoff should be unblocked.

**Comment by josefree** [comment]
![image](https://user-images.githubusercontent.com/18413981/234781669-02992ad2-7cd0-4a73-a39a-e155649abd72.png)

I see the gap here. Currently, the service team is working on **Private Preview** or **limited Public Preview**, their spec pull request is into **RPSaaSMaster** branch of the private spec repo.   They should initiate the flow with target phase to be Private Preview.  The flow is defined to end up at API readiness.  

When the service moves on to **Public Preview** or **GA**,  they should initiate another spec pull request to merge into **main** branch of the public spec repo.  The specs could be exactly the same as previously created private preview version, but the version label should be with later date. Refer to https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md#api-versioning 
![image](https://user-images.githubusercontent.com/18413981/234784673-2344aedd-f135-4ee4-8e35-99040247e7c4.png)
In the release planner, they should initiate another flow with target phase to be Public Preview or GA.  Then the flow will include SDK release.  In the flow, they could initially associate a spec pull request in the private repo, but at then end of the API readiness signoff, they much click "Push to Public" button,  which is to generate another spec pull request into main branch of the public spec repo, and the associated spec pull request should be replaced with this pull request (if not, this is an issue of Release planner backend implementation).  "Push to Public" is required for SDK release because currently, SDK release pipeline does not support to release from the private repo. 

Let me know if this answers your question.

**Comment by ladonnaq** [comment]
@josefree thank you for the input, appreciate it.  I am going to close this issue.  I have opened 2 issues to track outstanding todos from this GitHub issue.   
https://github.com/Azure/azure-sdk-tools/issues/6066
https://github.com/Azure/azure-sdk-tools/issues/6068

**Comment by maririos** [comment]
Agree with @josefree comments that it looks like bugs in the Release planner. @ladonnaq could you please open issues for them?
